src/Controls/src/Core/Platform/GestureManager/GestureManager.Gtk.cs

```diff
@@ -1,17 +1,268 @@
+						oldRecognizers.CollectionChanged -= _collectionChangedHandler;
+					}
+				}
+
+				_element = value;
+
+				if (_element != null)
+				{
+					var view = _element as View;
+					if (view != null)
+					{
+						var newRecognizers = (ObservableCollection<IGestureRecognizer>)view.GestureRecognizers;
+						newRecognizers.CollectionChanged += _collectionChangedHandler;
+					}
+				}
+			}
+		}
+
+		public bool PreventGestureBubbling
+		{
+			get
+			{
+				return Element switch
+				{
+					Button => true,
+					CheckBox => true,
+					DatePicker => true,
+					Stepper => true,
+					Slider => true,
+					Switch => true,
+					TimePicker => true,
+					ImageButton => true,
+					RadioButton => true,
+					_ => false,
+				};
+			}
 		}
 
 		public void Dispose()
 		{
+			Dispose(true);
+			GC.SuppressFinalize(this);
+		}
+
+		private void Dispose(bool disposing)
+		{
+			if (_isDisposed)
+				return;
+
+			_isDisposed = true;
+
+			if (!disposing)
+				return;
+
+			if (_container != null)
+			{
+				_container.ButtonPressEvent -= OnContainerButtonPressEvent;
+			}
+
+			if (_element != null)
+			{
+				var view = _element as View;
+				if (view != null)
+				{
+					var oldRecognizers = (ObservableCollection<IGestureRecognizer>)view.GestureRecognizers;
+					oldRecognizers.CollectionChanged -= _collectionChangedHandler;
+				}
+			}
+
+			if (_control != null)
+			{
+				_control.ButtonPressEvent -= OnControlButtonPressEvent;
+			}
+
+			Container?.Destroy();
+			Container = null;
+		}
+
+		private void ModelGestureRecognizersOnCollectionChanged(object? sender, NotifyCollectionChangedEventArgs notifyCollectionChangedEventArgs)
+		{
+			UpdatingGestureRecognizers();
+		}
+
+		private void UpdatingGestureRecognizers()
+		{
+			var view = Element as View;
+			var gestures = view?.GestureRecognizers;
+
+			if (_container == null || gestures == null)
+				return;
+
+			_container.ButtonPressEvent -= OnContainerButtonPressEvent;
+
+			if (gestures.GetGesturesFor<TapGestureRecognizer>().Any() || gestures.GetGesturesFor<ClickGestureRecognizer>().Any())
+			{
+				_container.ButtonPressEvent += OnContainerButtonPressEvent;
+			}
+			else
+			{
+				if (_control != null && PreventGestureBubbling)
+				{
+					_control.ButtonPressEvent += OnControlButtonPressEvent;
+				}
+			}
 		}
+
+		private void OnContainerButtonPressEvent(object o, ButtonPressEventArgs args)
+		{
+			var button = args.Event.Button;
+			if (button != 1 && button != 3)
+			{
+				return;
+			}
+
+			var view = Element as View;
+
+			if (view == null)
+				return;
+
+			int numClicks = 0;
+			switch (args.Event.Type)
+			{
+				case Gdk.EventType.ThreeButtonPress:
+					numClicks = 3;
+					break;
+				case Gdk.EventType.TwoButtonPress:
+					numClicks = 2;
+					break;
+				case Gdk.EventType.ButtonPress:
+					numClicks = 1;
+					break;
+				default:
+					return;
+			}
+
+			// Taps or Clicks
+			if (button == 1)
```

Comment:
Can't we use smth like `ButtonsMask.Primary` instead of 1 here?

Reply:
Actually I see that 1 here means both tap and click

Reply:
Changed to use `ButtonsMask.Primary`

Reply:
time to click on "Resolve conversation"?